### ISSUE TITLE
Bug: swap currency benchmark fails with TokenPoolAlreadyExists

### ISSUE THREAD

**Issue by enthusiastmartin** [galacticcouncil/Basilisk-node#238]
On Basilisk master, if you run swap_currency benchmark in multi-payment pallet. - If fails with error TokenPoolAlreadyExists. 

` cargo run --release --features=runtime-benchmarks -- benchmark --chain=dev --steps=5 --repeat=20  --execution=wasm --wasm-execution=compiled --heap-pages=4096  --pallet=pallet_transaction_multi_payment  --extrinsic="swap_currency"`

However if you remove a call from this benchmark which adds currency to list of accepted currencies ( just comment it out )  - it is ok and benchmark completes without issues.

```rust

      swap_currency {
		let maker = funded_account::<T>("maker", 1);
		initialize_pool::<T>(maker, ASSET_ID, 1_000_000_000_000, Price::from(1))?;
		
		// MultiPaymentModule::<T>::add_currency(RawOrigin::Root.into(), ASSET_ID, Price::from(10))?;

		let caller = funded_account::<T>("caller", 2);
		MultiPaymentModule::<T>::set_currency(RawOrigin::Signed(caller.clone()).into(), ASSET_ID)?;

	}: { MultiPaymentModule::<T>::swap_currency(&caller, 1000)? }
	verify{
		assert_eq!(MultiPaymentModule::<T>::get_currency(&caller), Some(ASSET_ID));
		#[cfg(test)]
		assert_eq!(T::Currencies::free_balance(ASSET_ID, &caller), 9999689661666);
	}
```

I cannot comprehend why having this call (completely unrelated to xyk pool) in the benchmark can affect the check of pool existence.

**Comment by Roznovjak** [comment]
I have made some experiments and found out that the benchmark is for some unknown reason executed 3 times (even though I set `--repeat 1` argument). During the first run, `add_currency` throws error as expected but it looks like it's not handled correctly by the benchmark because the storage is not reverted and the second run continues. That's why we get `TokenPoolAlreadyExists`, the pool was already created in the first run.
